tests/test_transformations.py

```diff
@@ -0,0 +1,203 @@
+import copy
+import unittest
+from avalanche.benchmarks.datasets.dataset_utils import default_dataset_location
+from avalanche.benchmarks.utils.detection_dataset import DetectionDataset
+
+
+from avalanche.benchmarks.utils.transforms import (
+    MultiParamCompose,
+    MultiParamTransformCallable,
+    TupleTransform,
+)
+
+import torch
+from PIL import ImageChops
+from torch import Tensor
+from torch.utils.data import DataLoader
+from torchvision.datasets import MNIST
+from torchvision.transforms import ToTensor
+from torchvision.transforms.functional import to_tensor
+from PIL.Image import Image
+
+from tests.unit_tests_utils import get_fast_detection_datasets
+
+
+def pil_images_equal(img_a, img_b):
+    diff = ImageChops.difference(img_a, img_b)
+
+    return not diff.getbbox()
+
+
+def zero_if_label_2(img_tensor: Tensor, class_label):
+    if int(class_label) == 2:
+        torch.full(img_tensor.shape, 0.0, out=img_tensor)
+
+    return img_tensor, class_label
+
+
+def get_mbatch(data, batch_size=5):
+    dl = DataLoader(
+        data, shuffle=False, batch_size=batch_size, collate_fn=data.collate_fn
+    )
+    return next(iter(dl))
+
+
+class TransformsTest(unittest.TestCase):
+    def test_multi_param_transform_callable(self):
```

Comment:
is it possible to add some tests for the transformation flattening?

Reply:
Added! Adding a unit test was quite useful as transformations were not properly flattened 🙃